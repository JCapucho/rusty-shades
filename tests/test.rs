use rusty_shades::build_naga_ir;
use std::fs::read_to_string;

#[test]
fn insta_test() {
    insta::glob!("*.rsh", |file| {
        let code = read_to_string(file).unwrap();
        let res = build_naga_ir(&code);
        insta::assert_ron_snapshot!(res);
    });
}
