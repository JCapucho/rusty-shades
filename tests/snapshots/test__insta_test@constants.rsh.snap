---
source: tests/test.rs
expression: res
input_file: tests/constants.rsh
---
Ok(Module(
  header: Header(
    version: (1, 0, 0),
    generator: 1920299892,
  ),
  types: [
    Type(
      name: None,
      inner: Vector(
        size: Tri,
        kind: Uint,
        width: 4,
      ),
    ),
  ],
  constants: [
    Constant(
      name: None,
      specialization: None,
      inner: Scalar(
        width: 4,
        value: Uint(1),
      ),
    ),
    Constant(
      name: Some("TRUE_HELP"),
      specialization: None,
      inner: Composite(
        ty: 1,
        components: [
          1,
          1,
          1,
        ],
      ),
    ),
    Constant(
      name: None,
      specialization: None,
      inner: Scalar(
        width: 4,
        value: Uint(2),
      ),
    ),
    Constant(
      name: Some("FALSE_HELP"),
      specialization: None,
      inner: Composite(
        ty: 1,
        components: [
          3,
          3,
          3,
        ],
      ),
    ),
    Constant(
      name: Some("HELP"),
      specialization: None,
      inner: Composite(
        ty: 1,
        components: [
          3,
          3,
          3,
        ],
      ),
    ),
  ],
  global_variables: [],
  functions: [],
  entry_points: {},
))
